Sources/Testing/Attachments/Test.Attachment.swift

```diff
@@ -129,19 +148,22 @@ extension Test.Attachment {
   /// value cannot be encoded and an error is thrown, that error is recorded as
   /// an issue in the current test and the resulting instance of
   /// ``Test/Attachment`` is empty.
+#if !SWT_NO_LAZY_ATTACHMENTS
   @_disfavoredOverload
+#endif
   public init(
     _ attachableValue: borrowing some Test.Attachable & ~Copyable,
     named preferredName: String? = nil,
     sourceLocation: SourceLocation = #_sourceLocation
   ) {
+    let preferredName = preferredName ?? Self.defaultPreferredName
     var proxyAttachable = _AttachableProxy()
     proxyAttachable.estimatedAttachmentByteCount = attachableValue.estimatedAttachmentByteCount
 
     // BUG: the borrow checker thinks that withErrorRecording() is consuming
     // attachableValue, so get around it with an additional do/catch clause.
```

Comment:
I'm wondering if we ought to propagate caught Errors in this initializer? Unlike the other one, where encoding the attachment is deferred, and it does make sense to catch it and record an Issue, this one might be better to propagate and let the test author decide how to handle it.

Reply:
I thought about it, but my reasoning was that:

1. Errors adding attachments are probably going to be pretty rare;
2. You'd have to `try` for non-sendable or move-only attachments, but not if the attachment is sendable and copyable, which is inconsistent and potentially confusing; and
3. An error that occurs when creating an attachment is _probably_ not a test failure, but an infrastructure failure, so it should not cause a test to halt execution—which it would unless you wrote `try?`, in which case we lose the error.